src/Datadog.Trace.ServiceFabric/Remoting.cs

```diff
@@ -0,0 +1,373 @@
+        // to generate [IgnoresAccessChecksToAttribute] and generate reference assemblies where they are public
+        private static readonly Datadog.Trace.Vendors.Serilog.ILogger Log = Datadog.Trace.Logging.DatadogLogging.GetLogger(typeof(Remoting));
+
+        private static int _firstInitialization = 1;
+        private static bool _initialized;
+        private static string? _clientAnalyticsSampleRate;
+        private static string? _serverAnalyticsSampleRate;
+
+        /// <summary>
+        /// Start tracing Service Remoting requests.
+        /// </summary>
+        public static void StartTracing()
+        {
+            // only run this code once
+            if (Interlocked.Exchange(ref _firstInitialization, 0) == 1)
+            {
+                // cache settings
+                _clientAnalyticsSampleRate = GetAnalyticsSampleRate(Tracer.Instance, enabledWithGlobalSetting: false)?.ToString(CultureInfo.InvariantCulture);
+                _serverAnalyticsSampleRate = GetAnalyticsSampleRate(Tracer.Instance, enabledWithGlobalSetting: true)?.ToString(CultureInfo.InvariantCulture);
+
+                // client events
+                ServiceRemotingClientEvents.SendRequest += ServiceRemotingClientEvents_SendRequest;
+                ServiceRemotingClientEvents.ReceiveResponse += ServiceRemotingClientEvents_ReceiveResponse;
+
+                // server events
+                ServiceRemotingServiceEvents.ReceiveRequest += ServiceRemotingServiceEvents_ReceiveRequest;
+                ServiceRemotingServiceEvents.SendResponse += ServiceRemotingServiceEvents_SendResponse;
+
+                // don't handle any events until we have subscribed to all of them
+                _initialized = true;
+            }
+        }
+
+        /// <summary>
+        /// Event handler called when the Service Remoting client sends a request.
+        /// </summary>
+        /// <param name="sender">The object that raised the event.</param>
+        /// <param name="e">The event arguments.</param>
+        private static void ServiceRemotingClientEvents_SendRequest(object? sender, EventArgs? e)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            GetMessageHeaders(e, out var eventArgs, out var messageHeaders);
+
+            var tracer = Tracer.Instance;
+            var span = CreateSpan(tracer, context: null, SpanKinds.Client, eventArgs, messageHeaders);
+
+            try
+            {
+                // inject propagation context into message headers for distributed tracing
+                if (messageHeaders != null)
+                {
+                    string samplingPriorityTag = span.GetTag(Tags.SamplingPriority);
+                    int? samplingPriority = int.TryParse(samplingPriorityTag, NumberStyles.None, CultureInfo.InvariantCulture, out int priority) ? priority : (int?)null;
```

Comment:
The value is stored as an enum internally. Can't you access it directly rather than have it converted to string then back to int?

`span.Context.TraceContext.SamplingPriority`

Reply:
Good catch. I will look into that.

Reply:
Fixed in 70f90a3e. It uses `[IgnoresAccessChecksTo]` which is a separate issue...

Reply:
Switched from `[IgnoresAccessChecksTo]` to `[InternalsVisibleTo]` in 0bc20cad, but note that it requires an update to `Datadog.Trace`.